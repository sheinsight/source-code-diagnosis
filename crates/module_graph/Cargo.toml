[package]
name = "module_graph"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
utils = { path = "../utils" }
beans = { path = "../beans" }
petgraph = "0.6.5"
bimap = "0.6.2"
camino = { workspace = true }
wax = "0.6.0"
rayon = { workspace = true }
crossbeam = "0.8.4"
napi-derive = { workspace = true }
napi = { workspace = true }
oxc_ast = { workspace = true }
oxc_resolver = { workspace = true }
oxc_parser = { workspace = true }
oxc_allocator = { workspace = true }
anyhow = { workspace = true }
log = { workspace = true }
serde_json = { workspace = true }
serde = { workspace = true }
oxc_span = { workspace = true }
pathdiff = { workspace = true }
tokio = { workspace = true, features = ["rt-multi-thread","test-util","macros"] }
pprof = { version = "0.12", features = ["flamegraph"] }
parking_lot = "0.12.1"
insta = { workspace = true }