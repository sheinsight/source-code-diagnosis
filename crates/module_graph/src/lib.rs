pub mod cycle;
pub mod edges;
pub mod graph;
pub mod macros;
pub mod model;
pub mod phantom_dependencies;
