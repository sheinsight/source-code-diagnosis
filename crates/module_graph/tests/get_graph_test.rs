use std::env::current_dir;

use module_graph::{edges::get_graph, model::JsArgs};

#[tokio::test]
async fn test_get_graph() -> anyhow::Result<()> {
  let current = current_dir()?
    .join("tests")
    .join("features")
    .join("get_graph");

  let args = JsArgs {
    cwd: current.to_string_lossy().to_string(),
    pattern: None,
    ignore: None,
    alias: None,
    modules: None,
  };

  let graphics = get_graph(args.into())?;

  let mut edges = graphics
    .graph
    .iter()
    .map(|edges| {
      let source = graphics.dictionaries.get(&edges.source).unwrap();
      let target = graphics.dictionaries.get(&edges.target).unwrap();
      format!("{} -> {}", source, target)
    })
    .collect::<Vec<_>>();

  edges.sort();

  insta::assert_snapshot!(edges.join("\n"));

  Ok(())
}
