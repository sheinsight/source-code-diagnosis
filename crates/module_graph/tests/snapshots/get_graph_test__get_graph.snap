---
source: crates/module_graph/tests/get_graph_test.rs
expression: "edges.join(\"\\n\")"
snapshot_kind: text
---
a.js -> ./c.js
a.js -> b.js
b.js -> a.js
b.js -> h/c.ts
h/c.ts -> b.js
